packages/arb-token-bridge-ui/src/components/common/NetworkSelectionContainer.tsx

```diff
@@ -0,0 +1,406 @@
+import Image from 'next/image'
+import {
+  CSSProperties,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState
+} from 'react'
+import { Chain } from 'wagmi'
+import { useDebounce } from 'react-use'
+import { XMarkIcon } from '@heroicons/react/24/outline'
+import { twMerge } from 'tailwind-merge'
+import { AutoSizer, List, ListRowProps } from 'react-virtualized'
+
+import { ChainId, getSupportedChainIds, isNetwork } from '../../util/networks'
+import { useAccountType } from '../../hooks/useAccountType'
+import { useIsTestnetMode } from '../../hooks/useIsTestnetMode'
+import { SearchPanel } from './SearchPanel/SearchPanel'
+import { SearchPanelTable } from './SearchPanel/SearchPanelTable'
+import { TestnetToggle } from './TestnetToggle'
+import { useArbQueryParams } from '../../hooks/useArbQueryParams'
+import {
+  panelWrapperClassnames,
+  onPopoverButtonClick,
+  onPopoverClose
+} from './SearchPanel/SearchPanelUtils'
+import { getBridgeUiConfigForChain } from '../../util/bridgeUiConfig'
+import { getWagmiChain } from '../../util/wagmi/getWagmiChain'
+import { useNetworks } from '../../hooks/useNetworks'
+
+type NetworkType = 'core' | 'orbit'
+
+enum ChainGroupName {
+  core = 'CORE CHAINS',
+  orbit = 'ORBIT CHAINS'
+}
+
+type ChainGroupInfo = {
+  name: ChainGroupName
+  description: string
+}
+
+const chainGroupInfo: { [key in NetworkType]: ChainGroupInfo } = {
+  core: {
+    name: ChainGroupName.core,
+    description: 'Chains managed directly by Ethereum or Arbitrum'
+  },
+  orbit: {
+    name: ChainGroupName.orbit,
+    description: 'Independent projects using Arbitrum technology.'
+  }
+}
+
+function ChainTypeInfoRow({
+  chainGroup,
+  style
+}: {
+  chainGroup: ChainGroupInfo
+  style: CSSProperties
+}) {
+  const { name, description } = chainGroup
+  const isCoreGroup = chainGroup.name === ChainGroupName.core
+
+  return (
+    <div
+      key={name}
+      style={style}
+      className={twMerge(
+        'px-6 py-3',
+        !isCoreGroup &&
+          'before:-mt-3 before:mb-3 before:block before:h-[1px] before:w-full before:bg-black/30 before:content-[""]'
+      )}
+    >
+      <p className="text-sm text-dark">{name}</p>
+      {description && <p className="mt-2 text-xs">{description}</p>}
+    </div>
+  )
+}
+
+function NetworkRow({
+  chainId,
+  style,
+  onClick,
+  close
+}: {
+  chainId: ChainId
+  style: CSSProperties
+  onClick: (value: Chain) => void
+  close: (focusableElement?: HTMLElement) => void
+}) {
+  const { network, nativeTokenData } = getBridgeUiConfigForChain(chainId)
+  const chain = getWagmiChain(chainId)
+  const [{ sourceChain }] = useNetworks()
+
+  function handleClick() {
+    onClick(chain)
+    close() // close the popover after option-click
+  }
+
+  function onKeyDown(event: React.KeyboardEvent<HTMLButtonElement>) {
+    if (event.key === 'Enter') {
+      handleClick()
+    }
+  }
+
+  return (
+    <button
+      onClick={handleClick}
+      onKeyDown={onKeyDown}
```

Comment:
Why not wrap it in a form?

Reply:
this is from the original header dropdown, i've removed it now because react-virtualized doesn't seem to support element focus within the `List`

it has to be looked into later